app/models/user.rb

```diff
@@ -41,6 +42,7 @@ class User < ApplicationRecord
   has_one :active_record_book, through: :active_participation, source: :record_book
 
   enum membership_type: { applicant: 0, member: 1, retired: 2 }
+  enum current_user_status: %w[active archived deleted]
```

Comment:
We can probably simplify this to just `status`.

What is the different between "archived" and "deleted"?

We should also throw a presence validator on this enum field so that saving stops before the DB breaks on a "NOT NULL" error.

Reply:
Yeah. That makes it sense.

This was a first iteration of the feature so I'm not really sure where I was going with this. But archived is a user that have participated in any record and deleted should probably just delete the user from the db. I think all in all, I was trying to avoid fully deleting the object in the off chance that they do comeback.

Will do. Not sure how to do that but I'll look into it.